dm/syncer/syncer.go

```diff
@@ -1464,11 +1478,30 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		}
 	}()
 
-	// some initialization that can't be put in Syncer.Init
 	fresh, err := s.IsFreshTask(runCtx)
 	if err != nil {
 		return err
-	} else if fresh {
+	}
+
+	// task command line arguments have the highest priority
+	// dm-syncer and other usage may not have a etcdCli, so we check it first
+	skipLoadMeta := false
+	if s.cli != nil {
```

Comment:
can we init a new etcd cli when s.cli is nil?

Reply:
> // dm-syncer and other usage may not have a etcdCli, so we check it first

what's your understanding of this?

Reply:
sorry i misreadning about `dm-syncer`.... i forgot some caller not use `NewSycner` to new a syncer object...